Add option to allocate mpi buffers in device memory (#979)
Hi-PACE/hipace: docs/source/run/parameters.rst

    Currently, this option only affects plasma operations (gather, push and deposition).
    The tile size can be set with ``plasmas.sort_bin_size``.

> * ``hipace.comms_buffer_on_gpu`` (`bool`) optional (default `false`)

Reviewer: We usually use `0` and `1` for booleans in the input file, could you make this consistent here?
```suggestion
* ``hipace.comms_buffer_on_gpu`` (`bool`) optional (default `0`)
```

    Whether the buffers used for MPI communication should be allocated on the GPU (device memory).
    By default they will be allocated on the CPU (pinned memory).

>     Setting this option to `1` is necessary to take advantge of GPU-Enabled MPI, however for this

Reviewer: It'll converge.
```suggestion
    Setting this option to `1` is necessary to take advantage of GPU-Enabled MPI, however for this
```

>     Setting this option to true is necessary to take advatige of GPU-Enabled MPI, however for this
>     additional enviroment variables need to be set depending on the system.

Reviewer: ```suggestion
    Setting this option to`1` is necessary to take advantge of GPU-Enabled MPI, however, for this,
    additional enviroment variables need to be set depending on the system.
```
